### PR TITLE
Help+LibManual: Support manpage section nesting

### PR DESCRIPTION
**Pull request by kleinesfilmroellchen**
This implements the actual motivating use case for LibManual: nested sections!

## Part 1: Preparatory cleanups

Some stuff left over after the recent LibManual port, as well as bugs found while implementing other parts of this PR.

### Help: Only toggle open/close for SectionNodes

The responsible code was actually casting everything to a SectionNode pointer, violating type safety all over the place and leading to frequent crashes. I'm surprised this was not exhibited before; I guess my recent changes made this bug surface.

### Help: Use array size instead of sizeof to determine number of sections

This was a relict when the sections were still a help-internal array.

### Help+LibManual: Make the children accessor fallible

This is convenient for the section node which might compute children on the fly.

### LibManual: Allow overriding a Node's path calculation

This is necessary for subclassing SectionNode, but generally allows more code to rely on path() virtual dispatch always finding the correct path regardless of the static type.

## Part 2: Remove "sections don't nest" assumptions

Surprisingly little code relied on this inside LibManual.

### Utilities: Resolve manpage paths more robustly in markdown-check

The path is now relative to the Serenity source directory, and later parts of the URL path are not simply discarded. This allows links into subsection man pages to be checked correctly.

### Help: Don't expand the whole tree of the selected node's parent

This now expands nodes unnecessarily with nested sections.

### Help: Detect clicked page correctly

index_from_path is the only remaining model index handling function that wasn't aware of subsections. After this change, clicked pages are resolved to a model index correctly, eliminating the weird subsection-expansion bugs from before.

## Part 3: Implement subsections and related neat Help features!

The main neat feature is that you see the page with the same name when directly clicking on a subsection. Removing the page's node in the tree view can be done separately as this PR is long enough as-is.

### Help+LibManual: Open sibling page for subsections

Clicking on a subsection now displays the sibling page, which is intended to be the main page for that section.

## Part 4: Reorganize the GML documentation into subsections

That was, like, the whole point of all of this. :poggie: 

I also took this opportunity to rewrite the documentation of the manpage system itself, so that subsections and their syntax are documented and we are not duplicating documentation across Help(1) and man(1).

### Base: Document manpage structure separately

This section is now removed from Help(1) and man(1).

### Base: Move GML documentation into subsection

That was the whole point of this endeavour :^)

### Base: Move GML Widget documentation to subsubsection

This is not only convenient, but also a subsection testcase :^)

### CODE REVIEW

File: Userland/Libraries/LibManual/SectionNode.cpp [hunk 1]
```diff
@@ -40,20 +42,36 @@ ErrorOr<void> SectionNode::reify_if_needed() const
...
+    Vector<Child> children;
 
-    Vector<String> page_names;
     while (dir_iter.has_next()) {
         LexicalPath lexical_path(dir_iter.next_path());
-        if (lexical_path.extension() != "md")
-            continue;
-        page_names.append(TRY(String::from_utf8(lexical_path.title())));
+        if (lexical_path.extension() != "md") {
+            dbgln("Non-md inode {}", lexical_path);
```
Comment (ADKaster) [inline Userland/Libraries/LibManual/SectionNode.cpp:57]: This dbgln and the one below will spam the console when browsing the man pages as-is, right? can we hide them behind a debug flag or remove?

File: Base/usr/share/man/man7/man.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+## Name
+
+man - SerenityOS manual system
+
+## Description
+
+The SerenityOS manual pages, or "man pages", document various parts of the operating system for users and developers. They are one of the two parts of the SerenityOS documentation.
+
+The other part of the SerenityOS documentation is the developer documentation, which can be found in the `Documentation` folder in the repository ([online link](https://github.com/SerenityOS/serenity/tree/master/Documentation)). This documentation is focused on setting up a SerenityOS installation and workflow, as well as contributing to its development.
```
Comment (ADKaster) [inline Base/usr/share/man/man7/man.md:9]: Usage of `This` in the second sentence is ambiguous.

```suggestion
The other part of the SerenityOS documentation is the developer documentation, which can be found in the `Documentation` folder in the repository ([online link](https://github.com/SerenityOS/serenity/tree/master/Documentation)). The developer documentation is focused on setting up a SerenityOS installation and workflow, as well as contributing to its development.
```

File: Base/usr/share/man/man7/man.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+Note that documentation might cover standardized topics (such as standard POSIX C library functions) or SerenityOS-specific extensions (such as custom file formats). There is always the possiblity of SerenityOS being non-spec-compliant, even if that is not documented.
```
Comment (ADKaster) [inline Base/usr/share/man/man7/man.md:11]: Not sure about the second sentence here. Perhaps an aspirational statement?

`SerenityOS intends to be spec-complaint with industry standard specifications. Non-compliance of with particular specification should be documented in the relevant sections.`

**Review by ADKaster — CHANGES_REQUESTED** [review]
These changes look really cool and should help reduce clutter in the man pages! 

Sorry it took so long to review. 

I have one code comment and two documentation comments, and this could probably use a rebase at this point.

File: Userland/Libraries/LibManual/SectionNode.cpp (hunk 1 above)
Comment (kleinesfilmroellchen) [reply Userland/Libraries/LibManual/SectionNode.cpp:57]: I will just remove them, it was to make sure that DirIterator picked up on exactly the things it should.

File: Base/usr/share/man/man7/man.md (hunk 3 above)
Comment (kleinesfilmroellchen) [reply Base/usr/share/man/man7/man.md:11]: Yeah, I just wanted to have this small word of warning that even though we provide POSIX stuff, we're non-compliant in sometimes obvious, sometimes subtle ways.

**Review by ADKaster — APPROVED** [review]

**Pull request merged**